lib/topological_inventory/schema/default.rb

```diff
@@ -123,6 +123,63 @@ def add_tagging_collection(model, manager_ref: [:source_ref])
         end
       end
 
+      def add_service_instances
+        add_collection(:service_instances) do |builder|
+          add_default_properties(builder)
+          add_default_values(builder)
+
+          save_block = lambda do |source, inventory_collection|
+            src_refs = inventory_collection.data.collect { |inventory_object| inventory_object.source_ref }
+            return if src_refs.blank?
+
+            # Saving Service Instances as usual
+            InventoryRefresh::SaveCollection::Base.send(:save_inventory, inventory_collection)
+
+            # Get running tasks
+            tasks_source_ref = Task.where(:state => 'running', :target_type => 'ServiceInstance', :source_id => source.id)
```

Comment:
@agrare is it worth adding index to `:target_type` if there is `[:target_type, :target_source_ref]`? I don't know which index is used by default (https://www.postgresql.org/docs/10/indexes-multicolumn.html) but it seems all of them can be used for subset of columns.